### DISCUSSION TITLE
`scoop update *` not  updating local bucket git repos

### DISCUSSION THREAD

**Discussion by fluffynuts** [ScoopInstaller/Scoop#5330] [Help and Questions]
## Bug Report
I noticed today pwsh telling me there was an update (from my 7.2.3 to 7.2.4). So naturally, being aware of https://github.com/ScoopInstaller/Main/issues/3572, I crack open powershell.exe, `pskill pwsh` and `scoop update *` and _nothing happens_. Scoop claims it's all up to date.

So I deleted `~/.config/scoop/config.json` and try again - I get the message that the last update time isn't set and scoop proceeds to update _nothing_

So I had to head manually off into each of the bucket repos under `~/scoop` and `git pull --rebase`, after which `scoop update *` had _heaps_ of updates to do.

(I did also try from within pwsh 7.2.3 before getting brute-force on it, and again, no updates applied)

#### Current Behavior
Doesn't pull git changes to buckets, so updates don't happen.

#### Expected Behavior
Should pull the git changes & perform updates

#### Possible Solution
I dunno, this used to work.

### System details

**Windows version:** win11

**OS architecture:** 64bit

**PowerShell version:** 5.1.22000.653

**Additional software:** [(optional) e.g. ConEmu, Git]

#### Scoop Configuration
```json
{
    "SCOOP_BRANCH":  "master",
    "lastupdate":  "2022-05-18T09:46:04.1056781+02:00",
    "SCOOP_REPO":  "https://github.com/ScoopInstaller/Scoop"
}
```

**Comment by fluffynuts** [comment]
see also:
![image](https://user-images.githubusercontent.com/4669340/168987331-673515f7-d303-4c52-8fa4-f21f346432a7.png)

**Comment by fluffynuts** [comment]
I also can't just re-install:
![image](https://user-images.githubusercontent.com/4669340/168987597-69efec84-42cb-43b7-a163-35fa924375d0.png)

**Comment by fluffynuts** [comment]
I had to manually remove `~/scoop/shims/scoop*` and then run `Invoke-WebRequest get.scoop.sh | Invoke-Expression`; now `scoop update * works` 😕

**Comment by rashil2000** [comment]
Unable to reproduce on powershell.exe, Windows 11

![image](https://user-images.githubusercontent.com/46838874/169080904-83720b2a-67bf-4c66-bbe2-11366b095c78.png)

**Comment by fluffynuts** [comment]
yeah, this is why I'm trying to transition to `winget` - "unable to repro, glhf" (and let's be honest, winget has it's own set of issues :| )

this is far from the first issue I've raised and it's always @rashil2000 responding with "sucks to be you"

no "gimme logs, I want to understand", no "do you have the scoop scripts that were failing", just "works on my machine". Not even a "can't repro, plz help". 

A package manager is all about empowering the user. You should read some gentoo bug reports to see how to opensource.

**Reply by darthwalsh** [reply]
Hey @fluffynuts I appreciate you're trying to make things better, but I don't think the attitude of calling out specific individuals is helping. If some open source tool doesn't solve your problems, it's fine to say that. But unless people are getting paid to maintain scoop I don't think we're entitled to a fix.

**Reply by fluffynuts** [reply]
Hi yourself

I suggested a solution (scoop should do what every user has to do - use PowerShell to update pwsh - or scoop should simply not manage pwsh, as it's now something which can be installed by Winget and kept up to date by windows update; or there's a third option where cleverness with an extra process could update pwsh without pwsh running).

I would have been happy to participate in the implementation, but it was clear that a specific person wasn't onboard.

I've raised PRs before (on other projects, to be clear) that languish and eventually rot. What a waste of time and effort! If there's no buy-in from the people who maintain the project, there's a good chance that I'd do that work for nothing. Hence my statements and my abandonment of scoop.

You may have just come in to see the tail end of the conversation, and it's easy to assume I'm just another entitled user, when the opposite is true: I used to heartily recommend scoop to people, I got involved who I could, reporting bugs, trying to find ways around them, feeling out if I could positively contribute. I've given up on that - too many other things to get done and Winget is working "ok" for me, though nothing is ever perfect.

I wish the project and its users well. I hope that some day, users can just update pwsh easily via scoop and never even have to see this issue again. Since the workaround is relatively simple (but arcane knowledge, let's be honest - it's not obvious that pwsh must be updated differently from other software), that could have been implemented in the year since this discussion - but it wasn't, so read into that what you will.

**Reply by darthwalsh** [reply]
I'm definitely more on the outside of this than you! I just opened a [new issue](https://github.com/ScoopInstaller/Main/issues/5255) linked from that thread for another footgun around `scoop.cmd` If I need to come back to this thread in the future, because something is still broken, I am planning on switching to winget for PWSH install! :) 

I'm particularly irked that I have come back to this issue thread multiple times trying to get the workaround to work. I haven't looked into using `winget` for managing PWSH -- you mentioned there are some downsides, is that something you've written about before? I don't recall reading specifics.

**Reply by fluffynuts** [reply]
The only downside is how long it takes for a new pwsh version to end up in winget - if you have the update reminder enabled in pwsh (it defaults on), you'll have a few days or a week when pwsh moans at you every time it starts up.

So what I've done now is:
1. install from winget
2. disable the update nags (set `POWERSHEL_UPDATECHECK` to `Off` as an environment variable - user or system-wide)
3. let windows update handle it - when there's an update to pwsh, it will come through good ol' win update (:

**Comment by rashil2000** [comment]
> this is far from the first issue I've raised and it's always @rashil2000 responding with "sucks to be you"

Would you mind showing me one more instance where I did this? 

> no "gimme logs, I want to understand", 

What logs? Scoop does not log anything 

> no "do you have the scoop scripts that were failing", just "works on my machine". 

What scripts? I ran the all the exact commands you gave me, and showed you

> Not even a "can't repro, plz help".

That's precisely what I did. Do you want me to spell out "plz help"?

**Comment by fluffynuts** [comment]
> > this is far from the first issue I've raised and it's always @rashil2000 responding with "sucks to be you"
> 
> Would you mind showing me one more instance where I did this?

in the included link about pwsh: https://github.com/ScoopInstaller/Main/issues/3572 where I, and other users, have expressed that the current situation should be remedied: having a package manager which can install a package once but not update it is totally unexpected. Your responses there are basically "nope"

> 
> > no "gimme logs, I want to understand",
> 
> What logs? Scoop does not log anything

if your scoop works and mine does not, then one avenue I would take if I were you would be to ask for copies of my scoop scripts. Perhaps something got broken in an update?

> 
> > no "do you have the scoop scripts that were failing", just "works on my machine".
> 
> What scripts? I ran the all the exact commands you gave me, and showed you

the scripts that _are_ scoop. Since a hard reinstall fixed it, there must have been something wrong, no?

> 
> > Not even a "can't repro, plz help".
> 
> That's precisely what I did. Do you want me to spell out "plz help"?

incorrect: you literally just said "unable to reproduce on powershell.exe, Windows 11". Which anyone would tell you does _not_ include any attempt to gather more information or fix the problem. It just says "works for me":
![image](https://user-images.githubusercontent.com/4669340/169223184-b9dfbab2-de69-4e2a-8774-372bb9d5a6a2.png)
